src/tools/imgtool/library.h

```diff
@@ -55,24 +55,27 @@ union filterinfo
 
 typedef void (*filter_getinfoproc)(uint32_t state, union filterinfo *info);
 
-struct imgtool_dirent
+namespace imgtool
 {
-	char filename[1024];
-	char attr[64];
-	uint64_t filesize;
-
-	time_t creation_time;
-	time_t lastmodified_time;
-	time_t lastaccess_time;
-
-	char softlink[1024];
-	char comment[256];
-
-	/* flags */
-	unsigned int eof : 1;
-	unsigned int corrupt : 1;
-	unsigned int directory : 1;
-	unsigned int hardlink : 1;
+	struct dirent
+	{
+		std::string filename;
+		std::string attr;
+		uint64_t filesize;
+
+		time_t creation_time;
+		time_t lastmodified_time;
+		time_t lastaccess_time;
```

Comment:
Should these be changed to something with more resolution, and should there be some kind of flag for "not available", and a flag for whether the source uses local or universal time?  We have other stuff that handles timestamps e.g. in the zip code, perhaps that's a place to look for inspiration?